### PR TITLE
feat: add ads label product card

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

Enable Ad label indicator in `ProductCard`. 

Since products are duplicated in the Product Gallery, it's important to clearly indicate which products are sponsored. This will help prevent users from finding the duplication confusing. Additionally, we want to provide an easier way to customize the sponsored label. To achieve this, we're adding the label to the ProductCard and enable text customization via the CMS.

|Before|After|
|-|-|
no-label|label

TODO: 

- [ ] Run cms sync in the main account to reflect the `sections.json` changes.
- [ ] Update `ProductCard` component doc with new tokens related to sponsored-label - I'll create a ticket when merging this PR
- [x] Verifying with IS team if sponsored products should be included in the count parameter (it's a bug or feature) - current scenario setting the query with `count=12` its returning 13 products.  [thread](https://vtex.slack.com/archives/C03L3CRCDC4/p1733413820077979?thread_ts=1733356825.821979&cid=C03L3CRCDC4)

## How it works?

Changing the label via CMS, the default text is `Sponsored`

|Admin|Store|
|-|-|
admin-cms|faststore-label

## How to test it?

1. Run the store locally
2. On `discovery.config.js`file, update `showSponsored` setting. 

```
 // Platform specific configs for API
  api: {
    ...
    showSponsored: true,
  }
```

3. Change via CMS, you can use my test account [fannyads](https://fannyads--storeframework.myvtex.com/admin/new-cms/faststore/)
   - On the Product List Page: Look for Product Card Configuration, and update the `Sponsored Label` input.
   - Update the text and save. Use the [CMS mode preview](https://developers.vtex.com/docs/guides/faststore/headless-cms-previewing-changes-in-development-mode) to visualize the change in the store.
   - You should be able to see the Label with the text.

### Starters Deploy Preview

[Preview Link - PLP](https://sfj-2e9ae6b--starter.preview.vtex.app/office?category-1=office&fuzzy=0&operator=and&facets=category-1%2Cfuzzy%2Coperator&sort=score_desc&page=0)
[Test PR](https://github.com/vtex-sites/starter.store/pull/628)

## References

[Jira ticket](https://vtex-dev.atlassian.net/jira/software/c/projects/SFS/boards/1051?selectedIssue=SFS-1607)
[Ad Network (Beta)](https://help.vtex.com/pt/tutorial/vtex-ad-network-beta--2cgqXcBuJmXN2livQvClur)

### CODE REVIEW

File: packages/components/src/molecules/ProductCard/ProductCardContent.tsx [hunk 1]
```diff
@@ -82,7 +90,9 @@ const ProductCardContent = forwardRef<HTMLElement, ProductCardContentProps>(
       onButtonClick,
       children,
       includeTaxes = false,
-      includeTaxesLabel = "Tax included",
+      includeTaxesLabel = 'Tax included',
+      isSponsored = true,
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/molecules/ProductCard/ProductCardContent.tsx:94]: my bad! It should be false, I forgot to update it. I'll do it 👍

File: packages/components/src/molecules/ProductCard/ProductCardContent.tsx [hunk 2]
```diff
@@ -65,6 +65,14 @@ export interface ProductCardContentProps extends HTMLAttributes<HTMLElement> {
    * Label to determine if the price includes taxes.
    */
   includeTaxesLabel?: string
+  /**
+   * Specifies if the displayed product is sponsored.
+   */
+  isSponsored?: boolean
+  /**
+   * Label to indicate if the product is sponsored.
+   */
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/molecules/ProductCard/ProductCardContent.tsx]: Great! Done [here](https://github.com/vtex/faststore/pull/2582/commits/3c4560a0a63e9a91499207fccecdb68fbd4dd16c) 🚀

**Review by lariciamota — APPROVED** [review]
LGTM! 🎉 
image

**Review by eduardoformiga — APPROVED** [review]
LGTM!
I am just wondering if it would be better to wait until the IS issue is addressed so that we can prevent gaps among ProductGrids. WDYT?
Screenshot 2024-12-06 at 17 57 02

File: packages/components/src/molecules/ProductCard/ProductCardContent.tsx [hunk 3]
```diff
@@ -62,9 +62,17 @@ export interface ProductCardContentProps extends HTMLAttributes<HTMLElement> {
...
   includeTaxes?: boolean
   /**
-   * Label to determine if the price includes taxes.
+   * Specifies the include taxes label, if taxes are included.
    */
   includeTaxesLabel?: string
+  /**
+   * Specifies if the displayed product is sponsored.
+   */
+  isSponsored?: boolean
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply packages/components/src/molecules/ProductCard/ProductCardContent.tsx]: niceee! [Done](https://github.com/vtex/faststore/pull/2582/commits/b63de2ca930324a947b3d9df5c457c00e3de59c6)!

**Pull request merged**